pkg/util/spill/agg_spill_util.go

```diff
@@ -0,0 +1,406 @@
+	retVal := *(*int8)(unsafe.Pointer(&buf[*pos]))
+	*pos += Int8Len
+	return retVal
+}
+
+// DeserializeUint8 deserializes int8 type
+func DeserializeUint8(buf []byte, pos *int64) uint8 {
+	retVal := *(*uint8)(unsafe.Pointer(&buf[*pos]))
+	*pos += Uint8Len
+	return retVal
+}
+
+// DeserializeInt32 deserializes int32 type
+func DeserializeInt32(buf []byte, pos *int64) int32 {
+	retVal := *(*int32)(unsafe.Pointer(&buf[*pos]))
+	*pos += Int32Len
+	return retVal
+}
+
+// DeserializeUint32 deserializes uint32 type
+func DeserializeUint32(buf []byte, pos *int64) uint32 {
+	retVal := *(*uint32)(unsafe.Pointer(&buf[*pos]))
+	*pos += Uint32Len
+	return retVal
+}
+
+// DeserializeUint64 deserializes uint64 type
+func DeserializeUint64(buf []byte, pos *int64) uint64 {
+	retVal := *(*uint64)(unsafe.Pointer(&buf[*pos]))
+	*pos += Uint64Len
+	return retVal
+}
+
+// DeserializeInt64 deserializes int64 type
+func DeserializeInt64(buf []byte, pos *int64) int64 {
+	retVal := *(*int64)(unsafe.Pointer(&buf[*pos]))
+	*pos += Int64Len
+	return retVal
+}
+
+// DeserializeFloat32 deserializes float32 type
+func DeserializeFloat32(buf []byte, pos *int64) float32 {
+	retVal := *(*float32)(unsafe.Pointer(&buf[*pos]))
+	*pos += Float32Len
+	return retVal
+}
+
+// DeserializeFloat64 deserializes float64 type
+func DeserializeFloat64(buf []byte, pos *int64) float64 {
+	retVal := *(*float64)(unsafe.Pointer(&buf[*pos]))
+	*pos += Float64Len
+	return retVal
+}
+
+// DeserializeMyDecimal deserializes float64 type
+func DeserializeMyDecimal(buf []byte, pos *int64) types.MyDecimal {
+	retVal := *(*types.MyDecimal)(unsafe.Pointer(&buf[*pos]))
+	*pos += types.MyDecimalStructSize
+	return retVal
+}
+
+// DeserializeTime deserializes Time type
+func DeserializeTime(buf []byte, pos *int64) types.Time {
+	retVal := *(*types.Time)(unsafe.Pointer(&buf[*pos]))
+	*pos += TimeLen
+	return retVal
+}
+
+// DeserializeDuration deserializes Duration type
+func DeserializeDuration(buf []byte, pos *int64) gotime.Duration {
+	retVal := *(*gotime.Duration)(unsafe.Pointer(&buf[*pos]))
+	*pos += DurationLen
+	return retVal
+}
+
+// DeserializeJSONTypeCode deserializes JSONTypeCode type
+func DeserializeJSONTypeCode(buf []byte, pos *int64) types.JSONTypeCode {
+	retVal := *(*types.JSONTypeCode)(unsafe.Pointer(&buf[*pos]))
+	*pos += JSONTypeCodeLen
+	return retVal
+}
+
+// DeserializeInterface deserializes interface type
+func DeserializeInterface(buf []byte, pos *int64) interface{} {
+	// Get type
+	dataType := int(buf[*pos])
+	*pos += InterfaceTypeLen
+
+	switch dataType {
+	case BoolType:
+		res := DeserializeBool(buf, pos)
+		return res
+	case Int64Type:
+		res := DeserializeInt64(buf, pos)
+		return res
+	case Uint64Type:
+		res := DeserializeUint64(buf, pos)
+		return res
+	case FloatType:
+		res := DeserializeFloat64(buf, pos)
+		return res
+	case StringType:
+		strLen := DeserializeInt64(buf, pos)
+		res := string(buf[*pos : *pos+strLen])
+		*pos += strLen
+		return res
```

Comment:
wrap this block into a DeserializeString ?

Reply:
> wrap this block into a DeserializeString ?

done